replica/database.cc

```diff
@@ -1824,11 +1824,10 @@ Future database::update_write_metrics(Future&& f) {
     return f.then_wrapped([this, s = _stats] (auto f) {
         if (f.failed()) {
             ++s->total_writes_failed;
-            try {
-                f.get();
-            } catch (const timed_out_error&) {
+            auto ep = f.get_exception();
+            if (is_timeout_exception(ep)) {
                 ++s->total_writes_timedout;
-                throw;
+                std::rethrow_exception(ep);
```

Comment:
Previously non-`timed_out_error` exceptions were not caught and were just propagated back, now they will trigger the assert.

Now we need to rethrow in the `else` case, and then the assert won't be needed.

Reply:
Right, sorry, typo.

Reply:
updated